test/tracerEq/test_consistency_2d.py

```diff
@@ -80,7 +81,16 @@ def run_tracer_consistency(constant_c=True, **model_options):
         tracer_init2d.interpolate(tracer_l + (tracer_r - tracer_l)*0.5*(1.0 + sign(x_2d - lx/4)))
 
     solver_obj.assign_initial_conditions(elev=elev_init, tracer=tracer_init2d)
-    solver_obj.iterate()
+
+    # time stepping
+    thetis_timestepper = solver_obj.create_iterator()
+    t_Thetis = solver_obj.simulation_time  # initial time
+    t_epsilon = 1.0e-5
+    while t_Thetis <= t_end - t_epsilon:
+        try:
+            t_Thetis = next(thetis_timestepper)
+        except StopIteration:
+            break
```

Comment:
```suggestion
    t_epsilon = 1.0e-5
    while True:
        try:
            t_Thetis = next(thetis_timestepper)
        except StopIteration:
            break
    assert t_Thetis>=t_end - t_epsilon
```
The loop you had either hits the StopIteration, or the while loop criterion - so one of them is redundant. I presume it's meant to hit the StopIteration. The assert that I suggest is then a way to check that it stops when we expect it.